test/case_teardown_failure/main.cpp

```diff
@@ -100,24 +100,21 @@ Case cases[] = {
 // Specification: Setup & Teardown ------------------------------------------------------------------------------------
 status_t greentea_setup(const size_t number_of_cases)
 {
-    MBED_HOSTTEST_TIMEOUT(5);
-    MBED_HOSTTEST_SELECT(default_auto);
-    MBED_HOSTTEST_DESCRIPTION(case setup failure test);
-    MBED_HOSTTEST_START("MBED_OS");
+    GREENTEA_START();
+    GREENTEA_SETUP(15, "default_auto");
 
     return verbose_test_setup_handler(number_of_cases);
 }
+
 void greentea_teardown(const size_t passed, const size_t failed, const failure_t failure)
 {
-    verbose_test_teardown_handler(passed, failed, failure);
-
     TEST_ASSERT_EQUAL(8, call_counter++);
     TEST_ASSERT_EQUAL(2, passed);
     TEST_ASSERT_EQUAL(1, failed);
     TEST_ASSERT_EQUAL(REASON_CASE_TEARDOWN, failure.reason);
     TEST_ASSERT_EQUAL(LOCATION_CASE_TEARDOWN, failure.location);
 
-    MBED_HOSTTEST_RESULT(true);
+    greentea_test_teardown_handler(passed, failed, failure);
```

Comment:
However, this will then also not report the expected failure, but that might not matter that much.